File: student-management-frontend/src/services/studentService.js
import api from './api';

export const studentService = {
    getAllStudents: async (page = 1, size = 10, search = '', sortBy = '') => {
        const response = await api.get('/students', {
            params: { page, size, search, sortBy }
        });
        return response.data;
    },
    
    getStudentById: async (id) => {
        const response = await api.get(`/students/${id}`);
        return response.data;
    },
    
    addStudent: async (student) => {
        const response = await api.post('/students', student);
        return response.data;
    },
    
    updateStudent: async (id, student) => {
        const response = await api.put(`/students/${id}`, student);
        return response.data;
    },
    
    deleteStudent: async (id) => {
        const response = await api.delete(`/students/${id}`);
        return response.data;
    }
};

File: student-management-system/src/main/java/com/example/student/service/StudentService.java
package com.example.student.service;

import com.example.student.dto.StudentRequest;
import com.example.student.model.Student;

import java.util.List;

public interface StudentService {
    void addStudent(StudentRequest request);
    List<Student> getAllStudents(int page, int size, String search, String sortBy);
    Student getStudentById(Integer id);
    void updateStudent(Integer id, StudentRequest request);
    void deleteStudent(Integer id);
}
